onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -531,6 +561,20 @@ static bool IsMovingSingleAxis(const std::vector<size_t>& permutations, size_t&
   return single_axis_moved;
 }
 
+bool IsReshape(const std::vector<size_t>& perm, const std::vector<int64_t>& input_dims) {
+  // A transposition only moving single dimension is equivalent to a reshape.
+  // Example: Shape=(1,1,1024,4096) -> perm=(2,0,3,1).
+  size_t last_permuted_axis = 0;
```

Comment:
This comment is a bit confusing as a 'single dimension' sounds like only one dim should move, but I believe the logic here is that as long as the dims with values > 1 stay in the same order, it's a reshape.

Reply:
Yes, that's what I intented to say. I'll rewrite.